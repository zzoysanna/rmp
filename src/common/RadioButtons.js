import React, { Component } from 'react';
import cn from 'classnames';
import './styles/radiobuttons.less';

export default class RadioButtons extends Component {
	constructor(props) {
		super(props);
		this.state = {
			active: this.props.active,
		};
	}

	setOption(option) {
		if(option !== this.state.active) {
			this.setState({
				active: option,
			});
			typeof this.props.onChange === 'function' && this.props.onChange(option);
		}
	}

	render() {
		const {options} = this.props;
		const {active} = this.state;
		return (
			<div className="radio-buttons">
				{options.map((option, i) => {
					return (
						<button
							key={i}
							className={cn({'active': active === option})}
							onClick={this.setOption.bind(this, option)}
						>
							{option}
						</button>
					);
				})}
			</div>
		);
	}
}
