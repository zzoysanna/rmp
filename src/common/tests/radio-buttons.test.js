import React from 'react';
import { shallow } from 'enzyme';
import RadioButtons from '../RadioButtons';

describe('RadioButtons', () => {

	beforeAll(() => {
	});

	it('should be render correctly', () => {
		const component = shallow(
			<RadioButtons options={['a', 'b']} active={'a'} onChange={jest.fn}/>
		);
		expect(component).toMatchSnapshot();
	});

	it('should change option', () => {
		const onChange = jest.fn();
		const component = shallow(
			<RadioButtons options={['a', 'b']} active={'a'} onChange={onChange}/>
		);
		component.find('button:not(.active)').simulate('click');
		expect(onChange).toHaveBeenCalled();
	});
});
