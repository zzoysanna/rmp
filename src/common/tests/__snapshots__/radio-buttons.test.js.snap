// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtons should be render correctly 1`] = `ShallowWrapper {}`;
